### ISSUE TITLE
Does not work with allure-pytest

### ISSUE THREAD

**Issue by jmatthews79** [browsertron/pytest-parallel#28]
When I attempt to use pytest-parallel it prevents the allure report from being generated. for instance if I run:
pytest --tests-per-worker 4 --alluredir=reports/allure

no report is generated. If i take the tests per worker argument out the report works. Any idea why this would prevent the report from being saved?

**Comment by jmatthews79** [comment]
looks like this is the same issue as the --junit one

**Comment by kevlened** [comment]
Hmm, I'd understand for `--workers`, but I'm not sure why it's not working for `--tests-per-worker`

**Comment by jmatthews79** [comment]
Let me rephrase that just to be clear. It is being generated, but its showing 0 tests executed.

**Comment by jmatthews79** [comment]
Hi @kevlened I was just curious if there was any plan to address the reporting bugs in the near future?

**Comment by jmatthews79** [comment]
@kevlened Completely understand, I hope I didn't come off as rude. I appreciate the work you did to create this. If I knew python other than to write some Selenium tests Id love to help. :)

**Comment by kevlened** [comment]
No worries (and thanks)!

**Comment by SizovIgor** [comment]
@kevlened Can you tell me your idea about a time for fix this issue (Bug) ?

**Comment by SizovIgor** [comment]
Please, look my changes, it is help me to work with allure-pytest in paralel.
I'm test with `--workers 2` and with `3,6,8,auto`.
But it is have a trouble with `--tests-per-worker` when it is greather than `1`. It is add N identical steps, where `N = tests-per-worker`


```diff --git a/pytest_parallel/__init__.py b/pytest_parallel/__init__.py
index 63d4175..8e71da4 100644
--- a/pytest_parallel/__init__.py
+++ b/pytest_parallel/__init__.py
@@ -8,6 +8,7 @@ import threading
 import queue as Queue
 from py._xmlgen import raw
 from multiprocessing import Manager, Process
+from allure_pytest.plugin import pytest_configure as original_pytest_configure
 
 __version__ = '0.0.9'
 
@@ -54,6 +55,7 @@ def run_test(session, item, nextitem):
 
 
 def process_with_threads(queue, session, tests_per_worker):
+    original_pytest_configure(allur_conf)
     threads = []
     for _ in range(tests_per_worker):
         thread = ThreadWorker(queue, session)
@@ -62,6 +64,9 @@ def process_with_threads(queue, session, tests_per_worker):
     [t.join() for t in threads]
 
 
+allur_conf = None
+
+
 class ThreadWorker(threading.Thread):
     def __init__(self, queue, session):
         threading.Thread.__init__(self)
@@ -69,6 +74,7 @@ class ThreadWorker(threading.Thread):
         self.session = session
 
     def run(self):
+        original_pytest_configure(allur_conf)
         while True:
             try:
                 index = self.queue.get_nowait()
@@ -87,6 +93,8 @@ class ThreadWorker(threading.Thread):
 
 @pytest.mark.trylast
 def pytest_configure(config):
+    global allur_conf
+    allur_conf = config
     workers = parse_config(config, 'workers')
     tests_per_worker = parse_config(config, 'tests_per_worker')
     if not config.option.collectonly and (workers or tests_per_worker):
```

**Comment by pmqs** [comment]
@SizovIgor did you get any further with getting `tests-per-worker` to work?

**Comment by kevlened** [comment]
The latest ([`0.1.0`](https://pypi.org/project/pytest-parallel/0.1.0/)) is quite a bit change that should resolve most reporting issues. It may solve this issue as well.

**Comment by aditya123456** [comment]
is this issue resolved? i am still facing this issue with 0.1.0

**Comment by EitanBe** [comment]
Hi @kevlened   
Thank you for your response.
I have same issue:
allure-pytest==2.8.6
pytest==6.0.1
pytest-parallel==0.1.0
Do you know when it can be resolve BTW, same with report portal. 
Thanks!

**Comment by talgabza** [comment]
I'm facing the same issue

Any solution/workaround for this problem?

**Comment by eldbud** [comment]
Facing same issue.

@kevlened - If i were to try and change the current situation, do you have a recommendation for a starting point? where in the repo should i investigate?

**Comment by kevlened** [comment]
Thanks for the interest. Here's what I'd do:

1) Repro
1) Do the brief [steps in CONTRIBUTING.md](https://github.com/browsertron/pytest-parallel/blob/master/CONTRIBUTING.md) and make sure the existing tests were working.
1) Write a failing [test case similar to the one for `pytest-html`](https://github.com/browsertron/pytest-parallel/blob/388e1941f3bc6ce8ce40eb95bbbe53e743e47226/tests/test_general.py#L202-L225). `pytest-allure` needs to be added to [tox.ini](https://github.com/browsertron/pytest-parallel/blob/388e1941f3bc6ce8ce40eb95bbbe53e743e47226/tox.ini#L11). I might skip this if I was unfamiliar with testing `pytest` plugins.
1) I'd start by trying [@SizovIgor's suggestion](https://github.com/browsertron/pytest-parallel/issues/28#issuecomment-535345123). The suggestion was prior to a pretty substantial change, but could be adapted.
1) If that didn't work, I'd investigate the `pytest-allure` repo to determine how the reports are generated. There's probably some data that `pytest-allure` depends on that `pytest-parallel` isn't sharing across threads. All of `pytest-parallel` is [a few hundred lines in this file](https://github.com/browsertron/pytest-parallel/blob/master/pytest_parallel/__init__.py).
1) If I still couldn't figure it out, I'd open a PR with the test case (if I wrote one), so others could give it a shot.

If you try it, let me know if you run into any issues.

**Comment by Phincle** [comment]
Facing same issue.

**Comment by door7474** [comment]
Facing same issue.

**Comment by Jabyn1031** [comment]
Any solution for this issue？

**Comment by b0mbays** [comment]
Same issue still. Any progress on this?
